akka-javasdk/src/main/java/akka/javasdk/agent/SessionMessage.java

```diff
@@ -28,7 +29,9 @@ static int sizeInBytes(String text) {
 
   String text();
 
-  record UserMessage(Instant timestamp, String text, String componentId) implements SessionMessage {
+  record UserMessageMetadata(String modelProvider, String modelName, double temperature, double topP, int topK, int reqMaxTokens) {}
+
+  record UserMessage(Instant timestamp, String text, String componentId, Optional<UserMessageMetadata> metadata) implements SessionMessage {
```

Comment:
system message is missing, because I'm not sure how we should add it, somehow repeating it each time feels wrong, but at the same time, it can change between calls...